File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>

using namespace std;

int M, S;


int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    
    cin >> M;
    for(int i = 0 ; i < M ; i++){
        string tmp;
        int idx;
        
        cin >> tmp;
        
        if(tmp == "add"){
            cin >> idx;
            idx = idx - 1;
            S |= (1<<idx); // idx번째 bit on
        }else if(tmp == "remove"){
            cin >> idx;
            idx = idx - 1;
            S &= ~(1<<idx); // idx 번째 bit off
        }else if(tmp == "check"){
            cin >> idx;
            idx = idx - 1;
            if(S & (1<<idx)) cout << 1 << '\n';
            else cout << 0 << '\n';
        }else if(tmp == "toggle"){
            cin >> idx;
            idx = idx - 1;
            if((S & (1<<idx))){ // idx번째가 켜져있다면 끄기
                S &= ~(1<<idx);
            } else{ // 꺼져있다면 키기.
                S |= (1<<idx);
            }
        }else if(tmp=="all"){
            S = 0xFFFFF;
        }else if(tmp=="empty"){
            S = 0;
        }
    }
    
    
    
    
    
}

File: 백준/Silver/11723. 집합/집합.py
import sys

N = int(sys.stdin.readline())

s = {} # Dic

for _ in range(N):
    tmp = sys.stdin.readline()
    if ' ' in tmp :
        cmd, v = tmp.split()
    else :
        cmd = tmp.rstrip()
    
    if cmd == 'add':
        if int(v) not in s or s[int(v)] == False:
            s[int(v)] = True
    elif cmd == 'remove':
        if int(v) in s and s[int(v)] == True :
            s[int(v)] = False
    elif cmd == 'check':
        print(1) if int(v) in s and s[int(v)]==True else print(0)
    elif cmd == 'toggle':
        if int(v) in s and s[int(v)]==True:
            s[int(v)]=False
        else:
            s[int(v)]=True
    elif cmd=='all':
        s = {i:True for i in range(1, 21)}
    elif cmd =='empty':
        s = {}
    
    
